controllers/btpoperator_controller.go

```diff
@@ -2026,3 +2139,84 @@ func (r *BtpOperatorReconciler) reconcileResourcesWithoutChangingCrState(ctx con
 		logger.Error(err, "resources reconciliation failed")
 	}
 }
+
+func (r *BtpOperatorReconciler) restartOperator(ctx context.Context, inNamespace string, logger *logr.Logger) error {
+	logger.Info(fmt.Sprintf("restarting of deployment: %s in %s started", DeploymentName, inNamespace))
+
+	clusterIdSecret := &corev1.Secret{
```

Comment:
Should be extracted to function with the proper responsibility - deletion of the secret.